pytorch_lightning/core/lightning.py

```diff
@@ -1591,9 +1595,13 @@ def to_torchscript(
                 # if no example inputs are provided, try to see if model has example_input_array set
                 if example_inputs is None:
                     example_inputs = self.example_input_array
+                # dicts are not supported, so show the user an error; not raising an error to show the original error
+                if isinstance(example_inputs, Mapping):
+                    log.error("`example_inputs` should be a Tensor or a tuple of Tensors,"
+                              f" but got {type(example_inputs)}.")
                 # automatically send example inputs to the right device and use trace
-                torchscript_module = torch.jit.trace(func=self.eval(), example_inputs=example_inputs.to(self.device),
-                                                     **kwargs)
+                example_inputs = move_data_to_device(example_inputs, device=self.device)
```

Comment:
```suggestion
                example_inputs = self.transfer_batch_to_device(example_inputs, device=self.device)
```
if someone overwrites this hook then we should use that only.

Reply:
good one!